### PR TITLE
Changed electric chair shock delay from 5s to 2s

### PR DESCRIPTION
**Pull request by Jujumatic**
Wanted to change this up because if you're stuck on an electric chair you're fucked already, so we might as well be able to spam the shock for a meme after you are inevitably dead. Did this change because I found out how to make electric chairs and I was upset that I couldn't spam electrocute my monkey victim.

:cl:
tweak: Changed electric chair's shock delay from 5 seconds down to 2 seconds. Having bodies being constantly shocked is more !!FUN!!
/:cl:

### CODE REVIEW

**Comment by Jujumatic** [comment]
Awe fuck this is non modular, gimme one second

**Comment by Jujumatic** [comment]
done, thought the electric chair was a hippie thing so i didnt check

**Comment by Jujumatic** [comment]
i forgot to disable the electric chair in the tg tree, bah

**Comment by Jujumatic** [comment]
oh god i'm fucking this up horribly, one sec

**Comment by Jujumatic** [comment]
fixed this up now pls merge so i can meme

File: hippiestation/code/game/objects/structures/electricchair.dm [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/obj/structure/chair/e_chair
+	name = "electric chair"
+	desc = "Looks absolutely SHOCKING!"
+	icon_state = "echair0"
+	var/obj/item/assembly/shock_kit/part = null
+	var/last_time = 1
+	item_chair = null
+
+/obj/structure/chair/e_chair/New()
+	..()
```
Comment (praisenarsie) [inline hippiestation/code/game/objects/structures/electricchair.dm]: Initialize() & . = ..()

Comment (Jujumatic) [reply hippiestation/code/game/objects/structures/electricchair.dm]: what will this do?

Comment (praisenarsie) [reply hippiestation/code/game/objects/structures/electricchair.dm]: better version of New()

**Comment by JohnGinnane** [comment]
@Jujumatic why do we need this?

**Comment by Jujumatic** [comment]
bc i wanna spam shock the corpses in the electric chair bc is funny xxDDDD

File: hippiestation/code/game/objects/structures/electricchair.dm [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		var/obj/structure/chair/C = new /obj/structure/chair(loc)
+		W.play_tool_sound(src)
+		C.setDir(dir)
+		part.forceMove(loc)
+		part.master = null
+		part = null
+		qdel(src)
+
+/obj/structure/chair/e_chair/proc/shock()
+	if(last_time + 10 > world.time)
```
Comment (JohnGinnane) [inline hippiestation/code/game/objects/structures/electricchair.dm]: Should probably use either a var (for admins to change) or a #define so it's easy to find and update

File: hippiestation.dme [hunk 3]
```diff
@@ -912,7 +912,6 @@
 #include "code\game\objects\structures\door_assembly.dm"
 #include "code\game\objects\structures\door_assembly_types.dm"
 #include "code\game\objects\structures\dresser.dm"
-#include "code\game\objects\structures\electricchair.dm"
```
Comment (JohnGinnane) [inline hippiestation.dme]: why did you remove this?

Comment (Jujumatic) [reply hippiestation.dme]: cause im making a hippie version, it got pissy when i tried modularising it because it said there were procs with the same name and didnt want to compile

Comment (JohnGinnane) [reply hippiestation.dme]: yeah because when you override a proc you don't need "proc" in the name. That's only when you're defining it for the first time (as it is being done in the /tg/ file)

Comment (Jujumatic) [reply hippiestation.dme]: so how would i write out the proc then?

Comment (JohnGinnane) [reply hippiestation.dme]: sorry, didn't see this

if you're overwriting an existing proc you do 
```
/path/to/obj/ProcName(args)
    . = ..() // This will call the previous definition so you don't need to copy the entire thing
    // stuff goes here
```

**Comment by JohnGinnane** [comment]
@Jujumatic not gonna lie, I noticed at least two rounds where you used the electric chair on lings/tators, and I get the feeling that that's what this is for, a buff to something you used recently. This seems a little biased? :thinking:

**Comment by Jujumatic** [comment]
The thing is, two shocks already put you into crit. Once you're put in that chair, you're not getting out alive anyway. I just wanted this for the meme effect afterwards, sort of like how you can put bodies on a conveyor belt going towards an electric fence and they get electrocuted every second or so.

File: hippiestation.dme (hunk 3 above)
Comment (Jujumatic) [reply hippiestation.dme]: but i'll still need to copypaste the proc itself because i want to change a value in it? doesn't that mean i can't just call the previous definition, since i won't be able to edit the values in there if i call it like that?

**Comment by Jujumatic** [comment]
anyway i fixed this up a bit, also due to the way the comparison with world.time works, this is actually on a 2 second timer. which is what i wanted to make anyway as i thought 1 second might be a bit TOO spammy

File: hippiestation/code/game/objects/structures/electricchair.dm [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/obj/structure/chair/e_chair/shock()
+	if(last_time + 10 > world.time)
```
Comment (GuyonBroadway) [inline hippiestation/code/game/objects/structures/electricchair.dm:2]: this is still only the one second...

Comment (Jujumatic) [reply hippiestation/code/game/objects/structures/electricchair.dm:2]: tested and it actually only works every 2 seconds, which is what i wanted anyway

**Comment by Carbonhell** [comment]
shit idea
-1

**Comment by earthmoonjune** [comment]
ye -1

**Comment by praisenarsie** [comment]
~~3~~ 2 strikes, you're out!
